cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -701,7 +701,19 @@ class SimpleArrayPlex
     {
     }
 
-    explicit SimpleArrayPlex(const shape_type & shape, DataType data_type);
+    explicit SimpleArrayPlex(const shape_type & shape, const double & value, const std::string & data_type)
+        : SimpleArrayPlex(shape, value, get_data_type_from_string(data_type))
+    {
+    }
+
+    explicit SimpleArrayPlex(const shape_type & shape, const std::shared_ptr<ConcreteBuffer> & buffer, const std::string & data_type)
+        : SimpleArrayPlex(shape, buffer, get_data_type_from_string(data_type))
+    {
+    }
+
+    explicit SimpleArrayPlex(const shape_type & shape, const DataType data_type);
+    explicit SimpleArrayPlex(const shape_type & shape, const double & value, const DataType data_type);
+    explicit SimpleArrayPlex(const shape_type & shape, const std::shared_ptr<ConcreteBuffer> & buffer, const DataType data_type);
```

Comment:
these three constructors follows the python binding implementation of `SimpleArray`